### PR TITLE
Move task limit from read pool to future pool

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <breezewish@pingcap.com>

###  What have you changed?

This PR moves the task limit functionality from read pool to future pool.

Extracted from https://github.com/tikv/tikv/pull/5220

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

### CODE REVIEW

File: cmd/src/setup.rs [hunk 1]
```diff
@@ -28,9 +28,8 @@ macro_rules! fatal {
 
 #[allow(dead_code)]
 pub fn initial_logger(config: &TiKvConfig) {
-    let log_rotation_timespan =
-        chrono::Duration::from_std(config.log_rotation_timespan.clone().into())
-            .expect("config.log_rotation_timespan is an invalid duration.");
+    let log_rotation_timespan = chrono::Duration::from_std(config.log_rotation_timespan.into())
```
Comment (AndreMouche) [inline cmd/src/setup.rs]: why?

Comment (breeswish) [reply cmd/src/setup.rs]: Because it has been implemented Copy.

**Comment by breeswish** [comment]
@siddontang @AndreMouche Fixed conflicts, please approve again, thanks!

**Pull request merged**